# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/platforms/apple/AppleVTDecoder.h
# Commit: eb882d1719d2
# Full Hash: eb882d1719d2db8d44a48599596a5aedbe899c69
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2020-05-15 21:52:36
# Description:
#   Bug 1636615: Work around Apple VT decoder saying it dropped a frame, and then decoding that frame anyway. r=jya
#   
#   We do this by keeping the unexpected frame in mReorderQueue, and outputting it later when the caller is expecting frames.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D75116
# ==============================================================================

diff -r 1b2022e31cd9 -r eb882d1719d2 dom/media/platforms/apple/AppleVTDecoder.h
--- a/dom/media/platforms/apple/AppleVTDecoder.h	Fri May 15 13:56:47 2020 +0000
+++ b/dom/media/platforms/apple/AppleVTDecoder.h	Thu May 14 23:45:26 2020 +0000
@@ -75,6 +75,7 @@
   RefPtr<DecodePromise> ProcessDrain();
   void ProcessShutdown();
   void ProcessDecode(MediaRawData* aSample);
+  void MaybeResolveBufferedFrames();
 
   void AssertOnTaskQueueThread() {
     MOZ_ASSERT(mTaskQueue->IsCurrentThreadIn());
